src/shop-redux-store.js

```diff
@@ -0,0 +1,301 @@
+import createStore from '../node_modules/@0xcda7a/redux-es6/es/createStore.js';
+
+const store = createStore(
+  (state, action) => {
+    let result = state;
+
+    switch (action.type) {
+      // window.location changed
+      case '_pathChanged':
```

Comment:
The conditions in this switch are conceptually distinct enough to be factored into their own reducers.